src/main/java/core/basesyntax/service/strategy/FruitStrategy.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.dto.FruitTransactionDto;
+import java.util.List;
+
+public class FruitStrategy {
+    private final List<OperationHandler> handlers;
+
+    public FruitStrategy(List<OperationHandler> handlers) {
+        this.handlers = handlers;
+    }
+
+    public OperationHandler findHandlerFor(FruitTransactionDto dto) {
+        for (OperationHandler handler : handlers) {
+            if (handler.isApplicable(dto)) {
+                return handler;
+            }
+        }
+        return null;
```

Comment:
Use `Map<Operation, OperationHandler>` to get the needed handler by operation enum with simply calling `map.get(operation)`
and initialize that map in main and pass it here in the constructor